src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/ProjectExtensions.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Buffers;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.CodeAnalysis;
+
+internal static class ProjectExtensions
+{
+    private const string GetTagHelpersEventName = "taghelperresolver/gettaghelpers";
+    private const string PropertySuffix = ".elapsedtimems";
+
+    /// <summary>
+    ///  Gets the available <see cref="TagHelperDescriptor">tag helpers</see> from the specified
+    ///  <see cref="Project"/> using the given <see cref="RazorProjectEngine"/>.
+    /// </summary>
+    public static ValueTask<ImmutableArray<TagHelperDescriptor>> GetTagHelpersAsync(
+        this Project project,
+        RazorProjectEngine projectEngine,
+        CancellationToken cancellationToken)
+        => project.GetTagHelpersAsync(projectEngine, NoOpTelemetryReporter.Instance, cancellationToken);
```

Comment:
Nit: I don't love how easy this makes it for us to resolve tag helpers without telemetry, as opposed to being able to easily see which method calls pass in `NoOpTelemetryReporter` but I don't feel too strongly about it, given we're unlikely to be adding any new callers any time soon.

Reply:
That's a fair point. I'll remove this overload and make callers pass an `ITelemetryReporter`.